packages/client/src/components/Login.js

```diff
@@ -27,6 +35,22 @@ const Login = () => {
     });
   }
 
+  //handle blur and validate
+  function handleBlur(e) {
+    const { name } = e.target;
+
+    if (name === 'email'){
+      emailSchema.validate({email: userObj.email})
+      .then(value => setErrors(prevErrors => ({...prevErrors, email: null})))
+      .catch(err => setErrors(prevErrors => ({...prevErrors, email: err.errors[0]})));
+    }
+    if  (name === 'password'){
+      passwordSchema.validate({password: userObj.password})
+      .then(value => setErrors(prevErrors => ({...prevErrors, password: null})))
+      .catch(err => setErrors(prevErrors => ({...prevErrors, password: err.errors[0]})));
+    }
```

Comment:
This isn't very DRY we repeat a lot of our code and if we were to continue adding fields, it'd very quickly become overwhelming, I wonder if there's some way we can clean it up. Something like:
```js
const schemas = { emailSchema, passwordSchema };
schemas[`${name}Schema`].validate({ [name]: userObj[name] })
  .then(_ => setErrors(prevErrors => ({ ...prevErrors, [name]: null })))
  .catch(({ errors }) => setErrors(prevErrors => ({ ...prevErrors, [name]: errors[0] })));
```

Reply:
Absolutely, this looks much better.  Hadn't seen the _ convention before either, just looked it up!